Kudu.Contracts/Settings/SettingsKeys.cs

```diff
@@ -42,10 +42,11 @@ public static class SettingsKeys
         public const string TouchWebConfigAfterDeployment = "SCM_TOUCH_WEBCONFIG_AFTER_DEPLOYMENT";
         public const string MaxRandomDelayInSec = "SCM_MAX_RANDOM_START_DELAY";
         public const string DockerCiEnabled = "DOCKER_ENABLE_CI";
-        public const string LinuxRestartAppContainerAfterDeployment = "SCM_RESTART_APP_CONTAINER_AFTER_DEPLOYMENT";
+        public const string RestartAppContainerAfterDeployment = "SCM_RESTART_APP_CONTAINER_AFTER_DEPLOYMENT";
```

Comment:
I would love to make this a more generic feature - not just for container.   It might be useful for MSDeploy or Windows scenario.

Reply:
Just to be sure I understand you correctly - You want to re-purpose the app setting to extend its scope to classic Windows apps too, while keeping the name unchanged? That would be a change in behavior for classic Windows apps as the default value of the app setting is assumed to be true. Is that fine?

Reply:
I thought it is off by-default - hence I'd love you to rename to be more generic to support any kind of windows applications as well.   Could we do this instead?

- Rename to more generic (not container specific).
- Lacking of such appsettings - for Windows, default is off and Container, default is on.

Reply:
I think, given that we will have to continue supporting SCM_RESTART_APP_CONTAINER_AFTER_DEPLOYMENT (container specific app-setting) for backward compatibility, adding a generic SCM_RESTART_APP_AFTER_DEPLOYMENT would be too confusing. Especially, given that the defaults would be different, it could be confusing what the expected behavior is for different combinations of the following: both app settings defined with conflicting values, one defined and one not defined, etc cases. 

I am thinking of the following.
- Re-purpose SCM_RESTART_APP_CONTAINER_AFTER_DEPLOYMENT (old app-setting) without renaming to work for Windows too. This means, the default behavior for Windows will change, but I guess a restart after a new deployment is the expected behavior anyway. If a particular app does not want this behavior, SCM_RESTART_APP_CONTAINER_AFTER_DEPLOYMENT can be defined explicitly to false.
- SCM_RESTART_APP_CONTAINER_AFTER_DEPLOYMENT would be a little misleading (because the name says 'container') if it is used for Windows apps too, but may be we can live with it.

What do you think?

---
In reply to: [334778351](https://github.com/projectkudu/kudu/pull/3050#discussion_r334778351) [](ancestors = 334778351)

Reply:
Sounds reasonable.   Let me know if you have squash and rebase - I can merge it in for you